source/utils/formatDate.mc

```diff
@@ -0,0 +1,62 @@
+import Toybox.Lang;
+import Toybox.Time;
+import Toybox.Time.Gregorian;
+
+module FormatDate {
+    module DisplayDateFormatType {
+        enum Enum {
+            DDMM = 0,
+            MMDD = 1
+        }
+    }
+
+    typedef FromatDateResult as {
+        :month as Lang.Number,
+        :day as Lang.Number,
+        :enMonthName as Lang.String,
+        :numMonthName as Lang.String,
+        :dayName as Lang.String
+    };
+
+    function formatDate(time as Time.Moment, format as Time.DateFormat?) as FromatDateResult {
+        var englishDays = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
+        var englishMonths = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
+        var numeralsMonths = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12"];
+        var date = Gregorian.info(time, format);
+
+        var month = date.month;
+        var day = date.day;
+        var enMonthName = englishMonths[month - 1];
+        var numMonthName = numeralsMonths[month - 1];
```

Comment:
Do we really need to convert a day to a string via an array? I thought a regular `date.month.toString()` would be enough

Reply:
FromatDateResult.month.toString() won't return different string results depending on the language?
I feel like the object of FromatDateResult.month is date.month.I am not sure about the details, and I need your advice. If there is no problem, I will delete numMonthName.